agent/consul/connect_ca_endpoint_test.go

```diff
@@ -573,28 +570,56 @@ func TestConnectCAConfig_Vault_TriggerRotation_Fails(t *testing.T) {
 			},
 		}
 	})
-	defer s1.Shutdown()
-
-	codec := rpcClient(t, s1)
-	defer codec.Close()
-
 	testrpc.WaitForTestAgent(t, s1.RPC, "dc1")
 
-	// Capture the current root.
-	{
-		rootList, _, err := getTestRoots(s1, "dc1")
-		require.NoError(t, err)
-		require.Len(t, rootList.Roots, 1)
-	}
-
-	cases := []struct {
+	// note: unlikely many table tests, the ordering of these cases does matter
+	// because any non-errored case will modify the CA config, and any subsequent
+	// tests will use the same agent with that new CA config.
+	testSteps := []struct {
 		name      string
-		configFn  func() (*structs.CAConfiguration, error)
+		configFn  func() *structs.CAConfiguration
 		expectErr string
 	}{
 		{
-			name: "cannot edit key bits",
-			configFn: func() (*structs.CAConfiguration, error) {
+			name: "allow modifying key type and bits from default",
```

Comment:
In this test and `"allow update that does not change key type or bits"` below, it's hard for me to parse what fields were changed (i.e. what the default values were at the start)

Reply:
I can add the default values to the initial config. Do you think that is enough to address this?

Reply:
Oh, I can also use the vars to specify the defaults, to make it more clear.

Reply:
I tried something else. I used an inline function to hide the common parts, and accept just the two values that are changing. Do you think this helps?

Reply:
Yeah, it's more clear this way. I'm still unclear what the difference between this test and `allow update that does not change key type or bits` is - the inputs are the same if I'm not mistaken?

Reply:
Ah I think I get it - first test case is `(default bits, default type -> rsa, 4096)`, whereas the last test has no diff but should still be a successful RPC call

Reply:
Yup, exactly!